### DISCUSSION TITLE
"Unsupported theme: theme requires true color support"

### DISCUSSION THREAD

**Discussion by sgon00** [helix-editor/helix#8031] [Troubleshooting] [answered]
Actually there are two problems related to true color support. 

1. I am using helix in a ssh session (both client and server are running debian). When I try to change the theme, it will report this error message: `"Unsupported theme: theme requires true color support"`

2. When I use helix in the `screen` app, the color is also not working. This happens in both ssh session and my local system. But the behaviors are different. It might because of different helix versions. In a ssh session, helix 23.05 will report `"Unsupported theme: theme requires true color support"`. In local system, helix 22.12 will simply mess up the color and entire screen becomes blue.

**Comment by pascalkuthe** [comment, marked as answer]
You can forcefully enable true color setting the `true-color` config option to `true`.  Gnu screen might not support true-colors properly

**Reply by sgon00** [reply]
Thanks for the tip. After enabling true-color to true, ssh session works. 
But gnu screen still does NOT work.

**Reply by sgon00** [reply]
I marked this as the answer because all the problems are caused by gnu screen. I switched to `zellij` and problem solved.

**Reply by gyreas** [reply]
one trick is to export `COLORTERM=truecolor` somehow before loading helix, perhaps in your `.bashrc`

**Reply by casperdcl** [reply]
Just for completeness/reference: https://docs.helix-editor.com/editor.html#editor-section

`hx ~/.config/helix/config.toml`

```toml
[editor]
true-color = true
```

**Comment by eminence** [comment]
I believe you can also set the environment variable `COLORTERM=truecolor` (at least this worked for me)

**Comment by lukepighetti** [comment]
Is there a list of themes that don't require truecolor?

**Reply by the-mikedavis** [reply]
The base16_ themes and term16_ themes shouldn't require true-color. Any theme that uses basic ANSI colors like `light-blue`, `yellow`, etc. and not custom colors (e.g. `#ff0000`) can be used without true-color support: https://github.com/helix-editor/helix/blob/fc97ecc3e3186b9dfe958869178bdb6b8cd7d8df/helix-view/src/theme.rs#L350-L356

**Reply by dawkrish** [reply]
It looks like some of the base_16 requires true-color

**Reply by mkf** [reply]
~~It might be helpful to some that one way to find a list of those should be to run in bash~~
```bash
( hx=`which hx` && cd ${hx%/*}/../lib/runtime/themes && grep -FL '"#' *.toml )
```
Actually, it just narrows down the list. Not all need to be specified as hex to match while remaining true color. Actually the ones that pass no `true-color` are just (manually checked)
* base16_terminal
* base16_transparent (that one in particular is helpful in palette trouble)
* term16_dark
* term16_light
* ttox
